### ISSUE TITLE
Receiving Data example not working as expected

### ISSUE THREAD

**Issue by garethjones99** [davidbutlerdesign/posistagedotnet#7]
Hello,

I have tried to do a simple WPF C# app to receive data using your example (specifically the orientation property), however it only seems to print the first value received. If i put a break point in the trackers updated callback and inspect the dictionary (e) it appears that the value is correct, however by the time it reaches the foreach loop, the log is always the first value received and never updates. The way i can work around this is by using the sender (s) and casting to a PsnClient which then correctly outputs my values.

Is this a bug with the callback function or is there something obvious i'm just not seeing ?

Here is an example which reproduces the error
```
using DBDesign.PosiStageDotNet;
using System;
using System.Linq;
using System.Net;
using System.Windows;

namespace PSNTest
{
    /// <summary>
    /// Interaction logic for MainWindow.xaml
    /// </summary>
    public partial class MainWindow : Window
    {
        PsnClient psnClient;
        public MainWindow()
        {
            InitializeComponent();

            var adapterIp = IPAddress.Parse("127.0.0.1");

            psnClient = new PsnClient(adapterIp);

            psnClient.TrackersUpdated += (s, e) =>
            {
                var client = (PsnClient)s;
                Console.WriteLine("This Works - " + client.Trackers.First().Value.Orientation);

                foreach (var t in e.Values)
                    Console.WriteLine("This is always the first value recieved - " + t.Orientation);
            };

            psnClient.StartListening();

            // Do something with the tracker data

            
        }

        private void Window_Closing(object sender, System.ComponentModel.CancelEventArgs e)
        {
            // When you're finished...
            psnClient.StopListening();

            // Don't forget to dispose!
            psnClient.Dispose();
        }
    }
}
```

**Comment by garethjones99** [comment]
To add to this i am receiving a packet every 100ms I believe it's a threading issue with the ReadOnlyDictionary which is a wrapper for the ConcurrentDictionary

**Comment by pixsperdavid** [comment]
Will take a look. Thanks for the report.

**Comment by pixsperdavid** [comment]
Fix available in v0.3.2
